libs/plugin/src/generators/convert-outputs/generator.ts

```diff
@@ -0,0 +1,316 @@
+import {
+	CodeBlockWriter,
+	Decorator,
+	Node,
+	Project,
+	WriterFunction,
+} from 'ts-morph';
+import { ConvertOutputsGeneratorSchema } from './schema';
+
+class ContentsStore {
+	private _project: Project = null!;
+
+	collection: Array<{ path: string; content: string }> = [];
+
+	get project() {
+		if (!this._project) {
+			this._project = new Project({ useInMemoryFileSystem: true });
+		}
+
+		return this._project;
+	}
+
+	track(path: string, content: string) {
+		this.collection.push({ path, content });
+		this.project.createSourceFile(path, content);
+	}
+}
+
+function trackContents(
+	tree: Tree,
+	contentsStore: ContentsStore,
+	fullPath: string,
+) {
+	if (fullPath.endsWith('.ts')) {
+		const fileContent =
+			tree.read(fullPath, 'utf8') || readFileSync(fullPath, 'utf8');
+		if (
+			!fileContent.includes('@Component') &&
+			!fileContent.includes('@Directive')
+		) {
+			logger.error(
+				`[ngxtension] "${fullPath}" is not a Component nor a Directive`,
+			);
+			return;
+		}
+
+		if (
+			fileContent.includes('@Output') &&
+			fileContent.includes('EventEmitter')
+		) {
+			contentsStore.track(fullPath, fileContent);
+		}
+	}
+}
+
+function getOutputInitializer(
+	propertyName: string,
+	decorator: Decorator,
+	initializer: string,
+): {
+	outputName?: string;
+	needsOutputFromObservableImport: boolean;
+	removeOnlyDecorator?: boolean;
+	writerFn: WriterFunction;
+} {
+	const decoratorArg = decorator.getArguments()[0];
+
+	// get alias if there is any from the decorator
+	let alias: string | undefined = undefined;
+	if (Node.isStringLiteral(decoratorArg)) {
+		alias = decoratorArg.getText();
+	}
+
+	// check if the initializer is not an EventEmitter -> means its an observable
+	if (!initializer.includes('EventEmitter')) {
+		// if the initializer is a Subject or BehaviorSubject
+		if (
+			initializer.includes('Subject') ||
+			initializer.includes('BehaviorSubject')
+		) {
+			// Before: @Output() outputFromSubject = new Subject();
+			// After: _outputFromSubject = outputFromObservable(this.outputFromSubject, { alias: 'outputFromSubject' });
+
+			return {
+				writerFn: (writer: CodeBlockWriter) => {
+					writer.write(`outputFromObservable`);
+					writer.write(`(this.${propertyName}`);
+					writer.write(`, { alias: ${alias ?? `'${propertyName}'`} }`);
+					writer.write(`);`);
+				},
+				outputName: `_${propertyName}`,
+				removeOnlyDecorator: true,
+				needsOutputFromObservableImport: true,
+			};
+		} else {
+			return {
+				writerFn: (writer: CodeBlockWriter) => {
+					writer.write(`outputFromObservable`);
+					writer.write(`(${initializer}`);
+					writer.write(`${alias ? `, { alias: ${alias} }` : ''}`);
+					writer.write(`);`);
+				},
+				needsOutputFromObservableImport: true,
+			};
+		}
+	} else {
+		let type = '';
+		if (initializer.includes('EventEmitter()')) {
+			// there is no type
+		} else {
+			const genericTypeOnEmitter = initializer.match(/EventEmitter<(.+)>/);
+			if (genericTypeOnEmitter.length) {
```

Comment:
needs to be `if (genericTypeOnEmitter?.length) {`

as `genericTypeOnEmitter` can be undefined 
with this migration run locally

Reply:
Thanks. On it.